feat(sdk): new section about bot-as-code (#71)
botpress/docs: for-developers/sdk/introduction.mdx

## Overview

> The Botpress SDK provides a robust and light-weight foundation for developing components within Botpress.

Reviewer: ```suggestion
The Botpress SDK provides a robust and lightweight foundation for developing components within Botpress.
```

## The HITL System

In Botpress, **HITL** (Human-in-the-Loop) may refer to an integration, an interface, or a plugin. The HITL system as a whole is a combination of these components working together to provide a human-in-the-loop experience.

The **HITL interface** defines the events and actions that an integration must implement to support human agent intervention. This interface is used by both the HITL plugin and the HITL integration.

The **HITL integration** is similar to integrations like Zendesk or Freshchat. It connects your bot to an external customer support service and implements the `hitl` interface by providing the required events and actions.

The **HITL plugin** contains the logic for human-in-the-loop functionality. It forwards messages to a human agent and allows the agent to respond. The plugin depends on the HITL interface, meaning it can work with any integration that implements it, including the HITL integration.

<CardGroup Cols={2}>
    <Card title="Read more" horizontal icon="book" href="/for-developers/sdk/interface/how-tos/implementing-hitl">

>         Learn how to implement the hitl interface in your integration

Reviewer: ```suggestion
        Learn how to implement the HITL interface in your integration
```

    </Card>
</CardGroup>

## Recommendations

> The Botpress SDK leverages TypeScript's strong typing system. We strongly recommend using an IDE that offers strong intellisense for Typescript such as [Visual Studio Code](https://code.visualstudio.com). When in doubt, the SDK typings should be used as a reference for the expected types and structures of various components.

Reviewer: ```suggestion
The Botpress SDK leverages TypeScript's strong typing system. We strongly recommend using an IDE that offers strong IntelliSense for Typescript such as [Visual Studio Code](https://code.visualstudio.com). When in doubt, the SDK typings should be used as a reference for the expected types and structures of various components.
```
